### PR TITLE
Add the --wait option to the task-group show command

### LINKED ISSUE #459: Add --wait option to "pulp task-group show"
## Summary
We can wait for a task-href. Would be nice to be able to wait-for a task-group to be complete. This would mean "monitor until all_tasks_dispatched=true and canceling/running/waiting all are 0"

### PR DESCRIPTION
**Pull request by lubosmj**
closes #459

### CODE REVIEW

File: CHANGES/459.feature [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Introduced the option ``--wait`` for the task-group show command. By using this option, details
+of a specific task-group will be shown only in case all related tasks will be finished.
```
Comment (mdellweg) [inline CHANGES/459.feature:2]:
```suggestion
of the task-group will be shown only after waiting for all related tasks to finish.
```
Please don't do the crazy wrapping here. Wrap whole sentences instead.

File: tests/scripts/pulp_file/test_acs.sh [hunk 2]
```diff
@@ -37,7 +37,12 @@ test "$(echo "$OUTPUT" | jq ".paths | length")" -eq 2
 # test refresh
 expect_succ pulp file acs refresh --name $acs
```
Comment (mdellweg) [inline tests/scripts/pulp_file/test_acs.sh:38]: maybe add `--background` here.

File: pulpcore/cli/core/task_group.py [hunk 3]
```diff
@@ -18,3 +28,49 @@ def task_group(ctx: click.Context, pulp_ctx: PulpContext) -> None:
 
 task_group.add_command(list_command())
 task_group.add_command(show_command(decorators=[href_option]))
```
Comment (mdellweg) [inline pulpcore/cli/core/task_group.py:30]: This ought to go now.

**Review by mdellweg — CHANGES_REQUESTED** [review]
Very nice. Just a few cosmetic changes.

File: pulpcore/cli/common/context.py [hunk 4]
```diff
@@ -296,7 +296,7 @@ def wait_for_task_group(self, task_group: EntityDefinition) -> Any:
                     time.sleep(1)
                     self.echo(".", nl=False, err=True)
                     task_group = self.api.call(
-                        "task_group_read", parameters={"task_group_href": task_group["pulp_href"]}
+                        "task_groups_read", parameters={"task_group_href": task_group["pulp_href"]}
```
[the remark this answers is not in the document]
Comment (lubosmj) [reply pulpcore/cli/common/context.py:299]: Yes, it is a bug fix. I think we do not have any tests for it. The method `wait_for_task_group` was not invoked anywhere.

[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/cli/common/context.py:299]: Good catch!

File: tests/scripts/pulp_file/test_acs.sh (hunk 2 above)
Comment (lubosmj) [reply tests/scripts/pulp_file/test_acs.sh:38]: Will "ERROUTPUT" be still accessible on the next line?

Comment (mdellweg) [reply tests/scripts/pulp_file/test_acs.sh:38]: It will only be overwritten on the next "expect_*"

Comment (mdellweg) [reply tests/scripts/pulp_file/test_acs.sh:38]: Or you mean that "--background" still tells you about the triggered taskgroup? I hope so.

**Review by mdellweg — APPROVED** [review]

**Pull request merged**